### DISCUSSION TITLE
Question about config.anonymous in Authentication plugins

### DISCUSSION THREAD

**Discussion by maximluzin** [Kong/kong#8143] [Help] [answered]
Hello. Authentication plugins have a field  "config.anonymous"(example https://docs.konghq.com/hub/kong-inc/key-auth/)

The documentation says that this field is "An optional string (Consumer UUID) value to use as an anonymous Consumer", but I see in code(https://github.com/Kong/kong/blob/master/kong/plugins/key-auth/handler.lua#L207) that plugins supports   id and username. Is bug in code or in documentation ?

**Comment by flrgh** [comment, marked as answer]
Hi @maximluzin :wave: 

You are correct: the bundled auth plugins support both consumer.id and consumer.username for looking up the anonymous user. Looks like we need to update some of our docs!
